src/controllers/confirm.company.controller.ts

```diff
@@ -16,11 +21,29 @@ export const get = async (req: Request, res: Response, next: NextFunction) => {
   }
 };
 
-export const post = async (req: Request, res: Response, next: NextFunction) => {
+export const post = async (req: Request, res: Response) => {
+  const company: CompanyProfile = await getCompanyProfile(req.query.companyNumber as string);
+  const session: Session = req.session as Session;
+  const eligibilityStatusCode: EligibilityStatusCode = await checkEligibility(session, company.companyNumber);
+  if (eligibilityStatusCode === EligibilityStatusCode.COMPANY_VALID_FOR_SERVICE) {
+    await createNewConfirmationStatement(req);
+    const url = TRADING_STATUS_PATH.replace(":companyNumber", company.companyNumber);
+    return res.redirect(url);
+  } else {
+    return displayEligibilityStopPage(res, eligibilityStatusCode );
+  }
```

Comment:
the post function will need to catch any errors thrown and return next(e), to prevent the page just timing out an event of an error. I'd wrap the whole function in a try catch